curiosrc/market/lmrpc/lmrpc.go

```diff
@@ -171,9 +170,8 @@ func ServeCurioMarketRPC(db *harmonydb.DB, full api.FullNode, maddr address.Addr
 		return xerrors.Errorf("net resolve: %w", err)
 	}
 
-	if len(laddr.IP) == 0 {
-		// set localhost
-		laddr.IP = net.IPv4(127, 0, 0, 1)
+	if len(laddr.IP) == 0 || laddr.IP.IsUnspecified() {
+		return xerrors.Errorf("market rpc server listen address must be a specific address, not %s (probably missing bind IP)", listen)
```

Comment:
This removes a footgun where it's possible to start the adapter bound to 0.0.0.0 + making nodes in the cluster connect to 0.0.0.0, which would obviously fail.